components/profile/playHistory.tsx

```diff
@@ -0,0 +1,237 @@
+    .join('&');
+  const { data: playHistory, isLoading } = useSWRHelper<PlayAttempt[]>(
+    `/api/user/play-history?${queryString}`,
+    undefined,
+    {
+      revalidateOnFocus: false,
+      revalidateOnReconnect: false,
+      refreshInterval: 0
+    },
+    !isPro(user)
+  );
+
+  let prevEndTime = 0;
+  let prevDate: string | null = null;
+
+  useEffect(() => {
+    setIsMobile(window.innerWidth < 768);
+    addEventListener('resize', () => {
+      setIsMobile(window.innerWidth < 768);
+    }, { passive: true });
+
+    return () => {
+      removeEventListener('resize', () => {
+        setIsMobile(window.innerWidth < 768);
+      });
+    };
+  }, []);
+  useEffect(() => {
+    if (playHistory) {
+      setAccumulatedPlayHistory(prev => [...prev, ...playHistory].filter((playAttempt, index, self) => {
+        return index === self.findIndex((t) => (
+          t._id.toString() === playAttempt._id.toString()
+        ));
+      })
+      );
+    }
+  }, [playHistory]);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      if (window.innerHeight + window.scrollY >= document.body.offsetHeight - 100) {
+        // Reached bottom
+        if (playHistory && playHistory.length > 0 && !isLoading) {
+          setCursor(playHistory[playHistory.length - 1]._id.toString());
+        }
+      }
+    };
+
+    window.addEventListener('scroll', handleScroll);
+
+    return () => window.removeEventListener('scroll', handleScroll);
+  }, [isLoading, playHistory]);
+
+  const handleSliderChange = (e: any) => {
+    const newValue = e.target.value;
+
+    if (minDurationMinutes === intermediateMinDuration) {
+      return;
+    }
+
+    setIntermediateMinDuration(newValue);
+    setMinDurationMinutes(intermediateMinDuration);
+    setAccumulatedPlayHistory([]); // Clear the play history
+    setCursor(null); // Reset the cursor
+  };
+
+  const display = accumulatedPlayHistory && accumulatedPlayHistory.map((playAttempt, index) => {
+    const level = playAttempt.levelId as EnrichedLevel;
+    let durationInbetween = null;
+
+    if (prevEndTime) {
+      durationInbetween = moment.duration(playAttempt.startTime - prevEndTime, 'seconds').humanize();
+    }
+
+    prevEndTime = playAttempt.endTime;
+
+    const currentDate = moment.unix(playAttempt.startTime).format('MMMM Do, YYYY');
+    const showDate = currentDate !== prevDate;
+
+    prevDate = currentDate;
+
+    const isLeftAligned = index % 2 === 0;
+
+    return (
+      <div
+        key={playAttempt._id.toString()}
+        className='flex flex-col gap-2'
+        style={{ position: 'relative' }}
+      >
+        {showDate && (
+          <div className={'md:text-lg text-xs absolute top-50% transform -translate-y-1/2 text-gray-500 sm:left-[-10%] lg:left-[-50%]'}>
+
+            {currentDate}
+          </div>
+        )}
+
+        { durationInbetween && <span className='text-gray-500 dark:text-gray-400 justify-center self-center'>{durationInbetween} later</span> }
```

Comment:
I think it may be cleaner to remove this and just have a big connected line going down the middle. IMO it is distracting vs just having the exact time for each play